modules/src/ics02_client/error.rs

```diff
@@ -206,5 +206,23 @@ define_error! {
                 format_args!("upgraded client height {0} must be at greater than current client height {1}",
                     e.upgraded_height, e.client_height)
             },
+
```

Comment:
IMHO, we should refrain from defining specific error types in `ics02_client::Error` and reuse the `ics02_client::Error::ImplementationSpecific` variant. We could also provide variants of this `ImplementationSpecific` error with different [`flex_error::ErrorSource`s](https://docs.rs/flex-error/0.4.2/flex_error/trait.ErrorSource.html). It would make sense to define errors that the module code could react to (for example by retrying, etc.), but it doesn't seem like there is any such case where this is happening currently.
What do you think @romac and @adizere?

Reply:
Thank you for your comment.
I agree with that. It is enough to reuse the `ics02_client::Error::ImplementationSpecific` so far because we can distinguish the absence of the module object or typical errors and other errors.